aiokafka/message_accumulator.py

```diff
@@ -58,16 +61,20 @@ def _is_full(self, key, value):
         """return True if batch does not have free capacity for append message
         """
         if self._relative_offset == 0:
-            # batch must contain al least one message
+            # batch must contain at least one message
             return False
         needed_bytes = MessageSet.HEADER_SIZE + Message.HEADER_SIZE
         if key is not None:
             needed_bytes += len(key)
         if value is not None:
             needed_bytes += len(value)
+        if needed_bytes > self._batch_size:
+            raise MessageSizeTooLargeError(
+                "Message is %d bytes which is larger than the maximum batch"
+                " size configured with max_batch_size" % needed_bytes)
         return self._buffer.tell() + needed_bytes > self._batch_size
 
-    def append(self, key, value, timestamp_ms):
+    def append(self, key, value, timestamp_ms, future=False):
```

Comment:
Lets move the `future` to constructor, that way we will have a way to differentiate between batches.

Reply:
I'm not following this. Can you show me what you mean?

Reply:
Add a flag `future_per_message=True` to constructor and remove it from `append` kwargs.
Code will be:
```python
if self._future_per_message:
    future = loop.create_future()
    self._message_futures.append(future)
    return future
else:
    return self._future
```

Reply:
Sorry, I should have been more explicit. Can you clarify what you meant by "that way we will have a way to differentiate between batches"?

Reply:
Currently, any batch is the same to sender routine, but if we add a flag like that we can see if it's an explicit user batch or one created from `send()` calls. Not that I can find any reason for that )

Reply:
Any update on moving this bool?